From feb233a8b92a646e73e891b4c226a75ba4fd06d3 Mon Sep 17 00:00:00 2001
From: Mark Harris <mark.hsj@gmail.com>
Date: Mon, 28 Dec 2020 11:47:53 -0800
Subject: [PATCH 1/4] picture.c: Fix signed shift undefined behavior

Fix ubsan report:
src/picture.c:136:17: runtime error: left shift of 255 by 24 places
cannot be represented in type 'int'
---
 src/picture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/picture.c b/src/picture.c
index ae40e5d..f3c3007 100644
--- a/src/picture.c
+++ b/src/picture.c
@@ -119,7 +119,7 @@ static int is_gif(const unsigned char *buf, size_t length){
 }
 
 #define READ_U32_BE(buf) \
-    (((buf)[0]<<24)|((buf)[1]<<16)|((buf)[2]<<8)|((buf)[3]&0xff))
+    (((opus_uint32)(buf)[0]<<24)|((buf)[1]<<16)|((buf)[2]<<8)|(buf)[3])
 
 /*Tries to extract the width, height, bits per pixel, and palette size of a
    PNG.
-- 
GitLab
